### PR TITLE
Visual improvements in WCSAxes

### PR DESCRIPTION
**Pull request by astrofrog**
### rcParams and styles

This adds support for enough rcParams that styles now look reasonable:

**Before (I'll just show once since you'll get the idea)**

seaborn

**After**

seaborn

seaborn-notebook

dark_background

### Tick label placement

Since ticks now face outwards by default, we need to be smarter about how we place the tick labels:

**Before**

ticks

**After**

ticks

### Other changes

Since I had to regenerate all reference images, I've removed the custom filenames, going instead with the pytest-mpl default, and I've ensured that we always use the default initial style.

Fixes https://github.com/astropy/astropy/issues/7485

### CODE REVIEW

File: astropy/visualization/wcsaxes/frame.py [hunk 1]
```diff
@@ -137,7 +138,7 @@ def _update_patch_path(self):
     def patch(self):
         self._update_patch_path()
         return PathPatch(self._path, transform=self.parent_axes.transData,
-                         facecolor='white', edgecolor='white')
+                         facecolor=rcParams['axes.facecolor'], edgecolor='white')
```
Comment (larrybradley) [inline astropy/visualization/wcsaxes/frame.py]: Why not also allow a custom edgecolor from `rcParams`?

File: astropy/visualization/wcsaxes/ticklabels.py [hunk 2]
```diff
@@ -128,23 +139,23 @@ def draw(self, renderer, bboxes, ticklabels_bbox):
                     if np.abs(self.angle[axis][i]) < 45.:
                         ha = 'right'
                         va = 'bottom'
-                        dx = - text_size * 0.5
+                        dx = - pad
```
Comment (larrybradley) [inline astropy/visualization/wcsaxes/ticklabels.py]: Why do you have a space after the minus sign (just curious)?  (and also in line 148).

File: astropy/visualization/wcsaxes/frame.py (hunk 1 above)
Comment (astrofrog) [reply astropy/visualization/wcsaxes/frame.py]: We draw the patch separately from the spines in WCSAxes - edgecolor is set above on line 102

File: astropy/visualization/wcsaxes/ticklabels.py (hunk 2 above)
Comment (astrofrog) [reply astropy/visualization/wcsaxes/ticklabels.py]: Not sure, will fix!

Comment (larrybradley) [reply astropy/visualization/wcsaxes/ticklabels.py]: and line 153 too...

**Pull request merged**